vmm/src/memory_manager.rs

```diff
@@ -1613,20 +1616,24 @@ impl MemoryManager {
         // the same time, calculate the total size needed for the contiguous
         // EPC region.
         let mut epc_region_size = 0;
+        let sgx_page_size = 1 << SGX_PAGE_SIZE_SHIFT;
         for epc_section in sgx_epc_config.iter() {
             if epc_section.size == 0 {
                 return Err(Error::EpcSectionSizeInvalid);
             }
-            if epc_section.size & 0x0fff != 0 {
+            if epc_section.size & (sgx_page_size - 1) != 0 {
                 return Err(Error::EpcSectionSizeInvalid);
             }
 
             epc_region_size += epc_section.size;
         }
 
         // Place the SGX EPC region on a 4k boundary between the RAM and the device area
-        let epc_region_start =
-            GuestAddress(((self.start_of_device_area.0 + 0xfff) / 0x1000) * 0x1000);
+        let epc_region_start = GuestAddress(
+            ((self.start_of_device_area.0 + sgx_page_size - 1) >> SGX_PAGE_SIZE_SHIFT)
+                << SGX_PAGE_SIZE_SHIFT,
```

Comment:
The use of division is clearer for the page boundary calculation. I don't think the use of the constant for the page size adds anything.

Reply:
Hi @rbradford Usually,  >> is faster than using /.  Though 4KB is a static size defined by SGX, but if we use a constant size, which will be better if SGX supports other size later.

Reply:
The rust compiler knows that division or multiplication by a constant multiple of 2 can be turned into shifts. If you want to extract the page size into a constant then that is fine, but I don't approve of the unnecessary change to use a shift and the extra constant required.

Reply:
@rbradford Let me remove the << >> operations. Thanks.